### PR TITLE
feat(issue-details): Add URL and breadcrumbs sidebar to issue replay when in fullscreen mode

### PR DESCRIPTION
**Pull request by malwilley**
Ref https://github.com/getsentry/sentry/issues/63157

Adds the URL bar and breadcrumbs sidebar to the fullscreen view. While doing so, extracted the fullscreen and sidebar toggle buttons into new common components.

The only replay-side changes aside from that is some filtering in the breadcrumbs to ensure that the `clipWindow` is applied to those items.

CleanShot 2024-01-25 at 16 22 30

### CODE REVIEW

File: static/app/components/replays/replayCurrentUrl.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ function ReplayCurrentUrl() {
 
   if (!replay || !url) {
     return (
-      <TextCopyInput size="sm" disabled>
+      <TextCopyInput aria-label="Current URL" size="sm" disabled>
```
Comment (ryan953) [inline static/app/components/replays/replayCurrentUrl.tsx:32]: Translate here and other labels in the file

File: static/app/views/replays/detail/breadcrumbs/index.tsx [hunk 2]
```diff
@@ -55,8 +55,15 @@ function Breadcrumbs() {
...
-  const frames = replay?.getChapterFrames();
+  const startTimestampMs =
+    replay?.getReplay()?.started_at?.getTime() ?? 0 + startTimeOffsetMs;
+  const frames = replay
+    ?.getChapterFrames()
+    .filter(
+      frame =>
+        frame.offsetMs >= startTimeOffsetMs &&
+        frame.offsetMs <= startTimeOffsetMs + durationMs
+    );
```
Comment (ryan953) [inline static/app/views/replays/detail/breadcrumbs/index.tsx:66]: Here's that re-filtering I was thinking would happen. getChapterFrames is memorized, and this list isn't now. I think the clipped knowledge needs to be inside the reader is we can.

File: static/app/components/replays/replayFullscreenButton.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import {Button} from 'sentry/components/button';
+import {IconContract, IconExpand} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import useIsFullscreen from 'sentry/utils/window/useIsFullscreen';
+
+type ReplayFullscreenButtonProps = {
```
Comment (ryan953) [inline static/app/components/replays/replayFullscreenButton.tsx:6]: `Props` is probably sufficient, it's a small file, no need to be wordy

File: static/app/components/replays/replaySidebarToggleButton.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import {Button} from 'sentry/components/button';
+import {IconChevron} from 'sentry/icons';
+import {t} from 'sentry/locale';
+
+type ReplaySidebarToggleButtonProps = {
```
Comment (ryan953) [inline static/app/components/replays/replaySidebarToggleButton.tsx:5]: ditto to `Props`

File: static/app/views/replays/detail/breadcrumbs/index.tsx (hunk 2 above)
Comment (malwilley) [reply static/app/views/replays/detail/breadcrumbs/index.tsx]: Good point - put this into into a useMemo.

I still agree that putting this into the reader is better. Once we get this feature complete enough to test and get feedback I'll circle back to my wip branch. If I can't get it working on the second round I'll reach out and see if you've got any ideas!

File: static/app/components/replays/replayFullscreenButton.tsx (hunk 3 above)
Comment (malwilley) [reply static/app/components/replays/replayFullscreenButton.tsx]: That's fair, I have a vs code expander that auto names the props this way but I'll name it Props to match the rest of the replays components

File: static/app/components/replays/replayCurrentUrl.tsx (hunk 1 above)
Comment (malwilley) [reply static/app/components/replays/replayCurrentUrl.tsx]: Good catch! Added

**Review by ryan953 — APPROVED** [review]

**Pull request merged**